resources/assert_lib.brs

```diff
@@ -60,3 +61,81 @@ function __formatError(errMessage)
         stack: {}
     }
 end function
+
+sub __deepEquals(actual, expected, error)
+    if deepEquals(actual, expected) = true then
+        m.__pass()
+    else
+        m.__fail(m.formatError(error))
+    end if
+end sub
+
+'utility function to compare two objects/associative arrays with nested data structures
+function deepEquals(left as object, right as object) as boolean
```

Comment:
Watch out!  These functions are in-scope in all unit tests executed by `roca`, so we should try to avoid names that may collide with a user's codebase.  Consider something like `__roca_deepEquals`, which I suspect most Roku codebases won't have defined 😃